### PR TITLE
Add docstring for torch.detach

### LINKED ISSUE #194885: Feedback about torch.detach
There is the following issue on this page: https://docs.pytorch.org/docs/2.13/generated/torch.detach.html#torch.detach

still no description in torch.detach item

cc @svekars @sekyondaMeta @AlannaBurke

### PR DESCRIPTION
**Pull request by gaurav-redhat**
## Summary
- Add missing docstring for `torch.detach` in `torch/_torch_docs.py`

Fixes #194885

## Test plan
- [ ] `python -c "import torch; print(torch.detach.__doc__)"`

### CODE REVIEW

File: torch/_torch_docs.py [hunk 1]
```diff
@@ -14777,6 +14777,25 @@ def merge_dicts(*dicts):
 """,
 )
 
+add_docstr(
```
Comment (soulitzer) [inline torch/_torch_docs.py]: nit on placement, let's put it where all the ops are sorted alphabetically, currently this is stuck between all the _copy ops used for functionalization.

File: torch/_torch_docs.py [hunk 2]
```diff
@@ -14777,6 +14777,25 @@ def merge_dicts(*dicts):
...
+    torch.detach,
+    r"""
+detach(input) -> Tensor
+
+Returns a copy of :attr:`input` that is detached from the autograd graph.
```
Comment (soulitzer) [inline torch/_torch_docs.py]: This doesn't actually copy input

Comment (soulitzer) [reply torch/_torch_docs.py]: Prefer: "Returns a new Tensor, detached from the current graph."

**Review by soulitzer — CHANGES_REQUESTED** [review]
Thanks for the PR, but how about something like this instead to link to existing docs:

```
  add_docstr(
      torch.detach,
      r"""
  detach(input) -> Tensor

  Returns a new Tensor, detached from the current graph.
  See :meth:`~Tensor.detach` for function description.

  Args:
      {input}
  """.format(**common_args),
  )
```

**Comment by gaurav-redhat** [comment]
Thanks for the review @soulitzer — addressed all three points in ff4a036:

- Moved torch.detach to alphabetical order (between dequantize and diag)
- Updated wording to "Returns a new Tensor, detached from the current graph."
- Replaced the duplicated description with a link to :meth:~Tensor.detach``
